python/lsst/daf/butler/registries/sql.py

```diff
@@ -415,9 +415,21 @@ def registerDatasetType(self, datasetType: DatasetType) -> bool:
         _, inserted = self._managers.datasets.register(datasetType)
         return inserted
 
-    def removeDatasetType(self, name: str) -> None:
+    def removeDatasetType(self, name: str | tuple([str, ...])) -> None:
         # Docstring inherited from lsst.daf.butler.registry.Registry
-        self._managers.datasets.remove(name)
+
+        for datasetTypeExpression in ensure_iterable(name):
+            datasetTypes = list(self.queryDatasetTypes(datasetTypeExpression))
+            if not datasetTypes:
+                _LOG.info(f"Dataset type {datasetTypeExpression} not defined")
+            else:
+                for datasetType in datasetTypes:
+                    self._managers.datasets.remove(datasetType.name)
+                    _LOG.info(f"Removed {datasetType.name}")
+
+    # def removeDatasetType(self, name: str) -> None:
```

Comment:
We will have to delete this commented out code. git history will always be around to tell us how things used to be.